lib/manager/terraform/extract.ts

```diff
@@ -23,52 +24,80 @@ export function getTerraformDependencyType(
     case 'provider': {
       return TerraformDependencyTypes.provider;
     }
+    case 'required_providers': {
+      return TerraformDependencyTypes.required_providers;
+    }
     default: {
       return TerraformDependencyTypes.unknown;
     }
   }
 }
 
+const dependencyBlockExtractionRegex = /^\s*(?<type>module|provider|required_providers)\s+("(?<lookupName>[^"]+)"\s+)?{\s*$/;
+const keyValueExtractionRegex = /^\s*(?<key>[^\s]+)\s+=\s+"(?<value>[^"]+)"\s*$/; // extracts `exampleKey = exampleValue`
+
 export function extractPackageFile(content: string): PackageFile | null {
   logger.trace({ content }, 'terraform.extractPackageFile()');
-  if (!content.includes('module "') && !content.includes('provider "')) {
+  if (
+    !content.includes('module "') &&
+    !content.includes('provider "') &&
+    !content.includes('required_providers ')
+  ) {
     return null;
   }
   const deps: PackageDependency[] = [];
   try {
     const lines = content.split('\n');
     for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
       let line = lines[lineNumber];
-      const terraformDependency = /^(module|provider)\s+"([^"]+)"\s+{\s*$/.exec(
-        line
-      );
+      const terraformDependency = dependencyBlockExtractionRegex.exec(line);
       if (terraformDependency) {
-        logger.trace(`Matched ${terraformDependency[1]} on line ${lineNumber}`);
-        const tfDepType: TerraformDependencyTypes = getTerraformDependencyType(
-          terraformDependency[1]
+        logger.trace(
+          `Matched ${terraformDependency.groups.type} on line ${lineNumber}`
+        );
+        const tfDepType = getTerraformDependencyType(
+          terraformDependency.groups.type
         );
-        const dep: PackageDependency = {
-          managerData: {
-            moduleName: terraformDependency[2],
-            terraformDependencyType: tfDepType,
-          },
-        };
+
         if (tfDepType === TerraformDependencyTypes.unknown) {
           /* istanbul ignore next */ logger.trace(
```

Comment:
At the moment this line is unreachable as we narrow down the lines in the package files using the regex above to only valid ones. 
https://github.com/renovatebot/renovate/pull/6550/files/95b42dca1e3005b12ebd8f1c8a594090fd564fe1#diff-052ccc017b4c41e0c777ddd62c40a547R36

I have added this in the first iteration of the manager because when more and more types are added we will have to eventually drop the explicit declaration of valid resource types. In that case this will be needed. 
Furthermore it is a security net if the regex is changed as this will prevent the crash of this manager.

Reply:
Ok, then we should use warn or error log level, because we shouldn't come here